corona-ide-main/src/main/java/com/coronaide/main/server/ProjectRestServer.java

```diff
@@ -28,26 +31,36 @@
 @RestController
 public class ProjectRestServer {
 
-	private final IProjectService projectService;
-	
-	@Inject
-	public ProjectRestServer(IProjectService projectService) {
-		this.projectService = projectService;
-	}
-	
-	@RequestMapping(value = "/projects", method = RequestMethod.GET)
-	public Collection<Project> getAllProjects() {
-		return projectService.getAll();
-	}
-	
-	@RequestMapping(value = "/projects", method = RequestMethod.POST)
-	public Project createProject(@RequestBody ProjectRequest request) throws IOException {
-		return projectService.create(request);
-	}
-	
-	@RequestMapping(value = "/projects", method = RequestMethod.DELETE)
-    public void deleteProject(@RequestBody ProjectDeleteRequest request) throws IOException {
+    private final IProjectService projectService;
+
+    @Inject
+    public ProjectRestServer(IProjectService projectService) {
+        this.projectService = projectService;
+    }
+
+    @RequestMapping(value = "/projects", method = RequestMethod.GET)
+    public Collection<Project> getAllProjects() {
+        return projectService.getAll();
+    }
+
+    @RequestMapping(value = "/projects", method = RequestMethod.POST)
+    public Project createProject(@RequestBody ProjectRequest request) throws IOException {
+        return projectService.create(request);
+    }
+
+    /**
+     * @deprecated Use {@link #deleteProject(UUID, Boolean)} instead
+     */
+    @Deprecated
+    @RequestMapping(value = "/projects", method = RequestMethod.DELETE)
+    public void deleteProjectRequest(@RequestBody ProjectDeleteRequest request) throws IOException {
         projectService.delete(request);
-	}
-	
+    }
+
+    @RequestMapping(value = "/projects/{projectId}", method = RequestMethod.DELETE)
+    public void deleteProject(@PathVariable("projectId") UUID projectId,
+            @RequestParam(value = "q", defaultValue = "false") Boolean deleteOnDisk) throws IOException {
```

Comment:
the query parameter for deleting probably shouldn't be `q`.
Maybe `deleteFromDisk` (also since that's what we call that boolean on other layers we should call it that here as well, not `deleteOnDisk`

Reply:
I apparently have significant muscle memory around "q"... I shall fix that